src/condor_negotiator.V6/matchmaker.cpp

```diff
@@ -3701,14 +3701,20 @@ Matchmaker::TransformSubmitterAd(classad::ClassAd &ad)
 		return true;
 	}
 
-	std::string authenticated_identity;
-	if (!ad.EvaluateAttrString(ATTR_AUTHENTICATED_IDENTITY, authenticated_identity)) {
+	std::string submitter_name;
+	if (!ad.EvaluateAttrString(ATTR_NAME, submitter_name)) {
 		return true;
 	}
+	auto last_at = submitter_name.find_last_of('@');
+	if (last_at == std::string::npos) {
+		return true;
+	}
+	auto accounting_domain = submitter_name.substr(last_at + 1);
+
 	std::vector<classad::ExprTree*> args;
 	args.reserve(2);
```

Comment:
I suspect that recent c++ libraries all implement the small vector optimization, so reserve(2) is probably not needed